proxy/src/coco/peer.rs

```diff
@@ -32,316 +39,400 @@ pub struct UserRevisions {
+        api.storage().reopen()?;
 
-        entity.try_map(|info| match info {
-            entity::data::EntityInfo::Project(info) => Some(info),
-            _ => None,
-        })
-    });
-    project_meta
-        .map(|project| {
-            with_browser(peer, &project.urn(), |browser| {
-                let stats = browser.get_stats()?;
-                Ok((project, stats).into())
-            })
-        })
-        .collect()
-}
+        Ok(())
+    }
 
-/// Get all [`UserRevisions`] for a given project.
-///
-/// # Parameters
-///
-/// * `peer` - the peer API we're interacting through
-/// * `owner` - the owner of this peer, i.e. the current user
-/// * `project_urn` - the [`RadUrn`] pointing to the project we're interested in
-///
-/// # Errors
-///
-///   * [`error::Error::LibradLock`]
-///   * [`error::Error::Git`]
-pub fn revisions(
-    peer: &PeerApi,
-    owner: &User,
-    project_urn: &RadUrn,
-) -> Result<NonEmpty<UserRevisions>, error::Error> {
-    let project = get_project(peer, project_urn)?;
-    let storage = peer.storage().reopen()?;
-    let repo = storage.open_repo(project.urn())?;
-    let mut user_revisions = vec![];
-
-    let (local_branches, local_tags) = with_browser(peer, &project.urn(), |browser| {
-        Ok((
-            source::branches(browser, Some(BranchType::Local))?,
-            source::tags(browser)?,
-        ))
-    })?;
-
-    if !local_branches.is_empty() {
-        user_revisions.push(UserRevisions {
-            identity: (peer.peer_id().clone(), owner.clone()).into(),
-            branches: local_branches,
-            tags: local_tags,
-        })
+    /// Our current peers [`PeerId`].
+    #[must_use]
+    pub fn peer_id(&self) -> PeerId {
+        let api = self.peer_api.lock().expect("unable to acquire lock");
+        api.peer_id().clone()
     }
 
-    for peer_id in repo.tracked()? {
-        let remote_branches = with_browser(peer, &project.urn(), |browser| {
-            source::branches(
-                browser,
-                Some(BranchType::Remote {
-                    name: Some(format!("{}/heads", peer_id)),
-                }),
-            )
-        })?;
+    /// Get the default owner for this `PeerApi`.
+    #[must_use]
+    pub fn default_owner(&self) -> Option<user::User<entity::Draft>> {
+        let api = self.peer_api.lock().expect("unable to acquire lock");
+
+        match api.storage().default_rad_self() {
+            Ok(user) => Some(user),
+            Err(err) => {
+                log::warn!("an error occurred while trying to get 'rad/self': {}", err);
+                None
+            },
+        }
+    }
 
-        let user = repo.get_rad_self_of(peer_id.clone())?;
+    /// Set the default owner for this `PeerApi`.
+    ///
+    /// # Errors
+    ///
+    ///   * Fails to set the default `rad/self` for this `PeerApi`.
+    pub fn set_default_owner(&self, user: User) -> Result<(), error::Error> {
```

Comment:
Does this only happen upon initialization?

Reply:
We use it in many place where set up state.